src/main/java/com/iridium/iridiumskyblock/managers/IslandManager.java

```diff
@@ -283,6 +284,69 @@ private void deleteIslandBlocks(Island island, World world, int y, CompletableFu
         }
     }
 
+    public CompletableFuture<Void> regenerateTerrain(Island island) {
+        return CompletableFuture.runAsync(() -> {
+            List<CompletableFuture<Void>> completableFutures = Arrays.asList(
+                    regenerateTerrain(island, getWorld(World.Environment.NORMAL)),
+                    regenerateTerrain(island, getWorld(World.Environment.NETHER)),
+                    regenerateTerrain(island, getWorld(World.Environment.THE_END))
+            );
+            completableFutures.forEach(CompletableFuture::join);
+        });
+    }
+
+    private CompletableFuture<Void> regenerateTerrain(Island island, World world) {
+        CompletableFuture<Void> completableFuture = new CompletableFuture<>();
+        if (world == null) {
+            completableFuture.complete(null);
+        } else {
+            Bukkit.getScheduler().runTask(IridiumSkyblock.getInstance(), () -> regenerateTerrain(island, world, world.getMaxHeight(), completableFuture, 0));
+        }
+        return completableFuture;
+    }
+
+    public void regenerateTerrain (Island island, World world, int y, CompletableFuture<Void> completableFuture, int delay) {
+
+        if (world == null) return;
+        Location pos1 = island.getPosition1(world);
+        Location pos2 = island.getPosition2(world);
+
+        if(!(IridiumSkyblock.getInstance().getDefaultWorldGenerator(world.getName(), null) instanceof VoidGenerator)) {
+
+            String worldName = "plugins/iridiumskyblock" + "/" + "regenWorlds" + "/" + world.getName() + "_regen";
+            if (Bukkit.getWorld(worldName) == null) {
+
+                WorldCreator worldCreator = new WorldCreator(worldName)
+                        .generator(IridiumSkyblock.getInstance().getDefaultWorldGenerator(world.getName(), null))
+                        .environment(world.getEnvironment())
+                        .seed(world.getSeed());
+
+                worldCreator.createWorld();
+            }
+
+            World regenWorld = Bukkit.getWorld(worldName);
```

Comment:
Is it creating this world in a single tick? does that cause any lag issues?

Reply:
uh probably, yeah